viite-backend/viite-main/src/main/scala/fi/vaylavirasto/viite/dynamicnetwork/LinkNetworkUpdater.scala

```diff
@@ -0,0 +1,646 @@
+   * @throws ViiteException if the given JSON cannot be deserialized to a list of [[LinkNetworkChange]]s
+   */
+  def parseJSONForLinkNetworkChanges(changeSetJSON: String, changeSetName: String): List[LinkNetworkChange] = {
+
+    def getSamingSet(changeSetJSON: String): List[LinkNetworkChange] = {
+      JsonMethods.parse(changeSetJSON)
+        .extract[List[LinkNetworkChange]] // We assume to get at least one object to process. If none possible, use .extractOpt
+      //.getOrElse(Seq.empty[LinkNetworkChange])
+    }
+
+    def reThrow(myMessage: String, t: Throwable) = {
+      val excStr = s"An exception when parsing dynamic link network change set $changeSetName"
+      throw ViiteException(s"$excStr:\r  $myMessage\r    ${t.getMessage}\r\r")
+    }
+
+    try {
+      changeSetJSON match {
+        case ""     => List[LinkNetworkChange]() // Nothing to process
+        case "[]"   => List[LinkNetworkChange]() // Nothing to process
+        case "{}"   => List[LinkNetworkChange]() // Nothing to process
+        case "[{}]" => List[LinkNetworkChange]() // Nothing to process
+        case _ => getSamingSet(changeSetJSON) // may throw ViiteException
+      }
+    }
+    catch {
+      case e: JsonParseException => reThrow("Incorrectly formatted JSON. Not a proper Link Network change JSON.",e)
+      case e: MappingException   => reThrow("Incorrectly formatted JSON. Not a proper Link Network change JSON.",e)
+      case t: Throwable          => reThrow("Probably incorrectly formatted JSON. Unexpected exception not considered:",t)
+    }
+  }
+
+  /**
+   * Wrapper for persistLinkNetworkChanges(List[LinkNetworkChange], ChangeSetMetaData),
+   * parsing ChangeSetMetaData from changeSetName, linkDataRetrievalTime, and linkGeomSource.
+   *
+   * @param changeSet List of link network changes ([[LinkNetworkChange]]) to be persisted
+   * @param linkDataRetrievalTime
+   * @param linkGeomSource Source from where the link data / geometries of the set have been
+   * @param changeSetName of type [[LinkNetworkChange]]
+   * @throws ViiteException if any of the change data is invalid, or incongruent, or there is no such change to be made within Viite
+   */
+  def persistLinkNetworkChanges(changeSet: List[LinkNetworkChange],
+                                changeSetName: String,
+                                linkDataRetrievalTime: DateTime,
+                                linkGeomSource: LinkGeomSource = LinkGeomSource.NormalLinkInterface
+                               ): Unit = {
+    val changeMetaData = ChangeSetMetaData(changeSetName, linkDataRetrievalTime, linkGeomSource)
+    persistLinkNetworkChanges(changeSet, changeMetaData)
+  }
+
+  /**
+   * Takes in a change set describing dynamic changes to the underlying link network,
+   * validates each change, and makes the corresponding changes to the Viite data structures,
+   * iff all changes are valid changes for Viite.
```

Comment:
if*

Reply:
Ok, liian kryptistä. 
Matemaattisehko merkintä "iff" eli "if and only if". Mutta pitänee purkaa auki, tai jättää tavalliseksi iffiksi. :)

Reply:
Hah, tai sit valistan ihmisiä, ja laitanki "iff (if and only if)"